File: ios/EverlinkModule.m
//
//  EverlinkModule.m
//  RNEverlinkSample
//
//  Created by Admin on 12/21/21.
//

#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(EverlinkModule, RCTEventEmitter)

RCT_EXTERN_METHOD(initialize:(NSString *)appIDKey)

RCT_EXTERN_METHOD(startListening:(BOOL)isOffline)

RCT_EXTERN_METHOD(stopListening)

RCT_EXTERN_METHOD(playVolume:(double)volume useLoudspeaker:(BOOL)useLoudspeaker)

RCT_EXTERN_METHOD(startEmitting)

RCT_EXTERN_METHOD(stopEmitting)

RCT_EXTERN_METHOD(startEmittingToken:(NSString *)token isOffline:(BOOL)isOffline)

RCT_EXTERN_METHOD(createNewToken:(NSString *)startDate)

RCT_EXTERN_METHOD(saveSounds:(NSArray *)tokensArray)

RCT_EXTERN_METHOD(clearSounds)

@end

File: ios/EverlinkModule.swift
//
//  EverlinkModule.swift
//  RNEverlinkSample
//
//  Created by Admin on 12/21/21.
//

import Foundation
import EverlinkBroadcastSDK

@objc(EverlinkModule)
class EverlinkModule: RCTEventEmitter, EverlinkConnectDelegate {
  
  var connectObj:Everlink?
  
  override func supportedEvents() -> [String]! {
    return ["onAudioCodeReceived", "onEverLinkError", "onMyTokenGenerated"]
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func initialize(_ appIDKey: String) -> Void {
    connectObj = Everlink(appID: appIDKey)
    connectObj?.delegate = self
  }
  
  func onAudioCodeReceived(token: String) {
    print(token)
    sendEvent(withName: "onAudioCodeReceived", body: token)
  }
  
  func onMyTokenGenerated(token: String, oldToken: String) {
    print("New token: " + token)
    print("Old token: " + oldToken)
    sendEvent(withName: "onMyTokenGenerated", body: ["oldToken": oldToken, "newToken": token])
  }
  
  @objc func startListening(_ isOffline: Bool) -> Void {
    connectObj?.startRecording(isOffline: isOffline)
  }
  
  @objc func stopListening() -> Void {
    connectObj?.stopRecording()
  }
  
  @objc func playVolume(_ volume: Double, useLoudspeaker: Bool) -> Void {
    connectObj?.playVolume(volume: Float(volume), loudspeaker: useLoudspeaker)
  }
  
  @objc func startEmitting() -> Void {
    connectObj?.startEmitting()
  }
  
  @objc func stopEmitting() -> Void {
    connectObj?.stopEmitting()
  }
  
  @objc func startEmittingToken(_ token: String, isOffline: Bool) -> Void {
    connectObj?.startEmittingToken(token: token, isOffline: isOffline)
  }
  
  @objc func createNewToken(_ startDate: String) -> Void {
    connectObj?.newToken(startDate: startDate)
  }
  
  @objc func saveSounds(_ tokensArray: [String]) -> Void {
    connectObj?.saveSounds(tokensArray: tokensArray)
  }
  
  @objc func clearSounds() -> Void {
    connectObj?.clearSounds()
  }
  
}
